src/main/java/org/terasology/simpleFarming/systems/VineAuthoritySystem.java

```diff
@@ -0,0 +1,264 @@
+            newVineDefinitionComponent.growthsTillRipe = vineDefinitionComponent.growthsTillRipe;
+            newVineDefinitionComponent.soilCategory = vineDefinitionComponent.soilCategory;
+            newVineDefinitionComponent.seedPrefab = seedItem.getParentPrefab().getName();
+            newVineDefinitionComponent.plantName = vineDefinitionComponent.plantName;
+            newVineDefinitionComponent.sapling = vineDefinitionComponent.sapling;
+            newVineDefinitionComponent.trunk = vineDefinitionComponent.trunk;
+            newVineDefinitionComponent.produce = vineDefinitionComponent.produce;
+            newVineDefinitionComponent.isSapling = true;
+
+            List<Vector3i> vineParts = new ArrayList<>();
+            vineParts.add(plantPosition);
+            newVineDefinitionComponent.parts = vineParts;
+
+            worldProvider.setBlock(plantPosition, saplingBlock);
+            EntityRef vineEntity = blockEntityRegistry.getBlockEntityAt(plantPosition);
+            vineEntity.addComponent(newVineDefinitionComponent);
+            scheduleVineGrowth(vineEntity, newVineDefinitionComponent);
+            inventoryManager.removeItem(seedItem.getOwner(), seedItem, seedItem, true, 1);
+        }
+    }
+
+    private void scheduleVineGrowth(EntityRef vineEntity, VineDefinitionComponent vineDefinitionComponent) {
+        delayManager.addDelayedAction(vineEntity, GROW_ACTION, vineDefinitionComponent.nextGrowth.getTimeRange());
+    }
+
+    /**
+     * Sends an "start new growth cycle" signal.
+     *
+     * @param event                   The delayed action trigger due to which this function is called
+     * @param entity                  The vine entity whose growth action is to be triggered
+     * @param vineDefinitionComponent The vine definition component of the vine entity
+     * @param blockComponent          The block component of the vine entity
+     */
+    @ReceiveEvent
+    public void scheduledVineGrowth(DelayedActionTriggeredEvent event, EntityRef entity, VineDefinitionComponent vineDefinitionComponent, BlockComponent blockComponent) {
+        if (event.getActionId().equals(GROW_ACTION)) {
+            entity.send(new OnVineGrowth());
+        }
+    }
+
+    /**
+     * Makes the vine grow when a growth signal is sent.
+     *
+     * @param event                   The growth signal that triggered the function
+     * @param entity                  The vine entity to be grown
+     * @param vineDefinitionComponent The vine definition component of the vine entity to be grown
+     * @param blockComponent          The block component of the vine entity
+     */
+    @ReceiveEvent
+    public void onPlantGrowth(OnVineGrowth event, EntityRef entity, VineDefinitionComponent vineDefinitionComponent, BlockComponent blockComponent) {
+        if (delayManager.hasDelayedAction(entity, GROW_ACTION)) {
+            delayManager.cancelDelayedAction(entity, GROW_ACTION);
```

Comment:
Yes it is. The onPlantGrowth method is triggered by a DelayedAction with the GROW_ACTION ID. This block cancels that delayed action because the growth cycle that it triggers has been successfully triggered.